ClangTidy warnings for `performance-enum-size`

```
/Users/jviotti/Projects/core/src/core/json/include/sourcemeta/core/json_value.h:48:14: error: enum 'ParsePhase' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
   48 |   enum class ParsePhase { Pre, Post };
      |              ^

/Users/jviotti/Projects/core/src/core/json/include/sourcemeta/core/json_value.h:51:14: error: enum 'Type' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
   51 |   enum class Type : std::uint8_t {
      |              ^

/Users/jviotti/Projects/core/src/core/jsonpointer/include/sourcemeta/core/jsonpointer_template.h:19:14: error: enum 'Wildcard' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
   19 |   enum class Wildcard { Property, Item, Key };
      |              ^

/Users/jviotti/Projects/core/src/core/jsonschema/frame.cc:13:12: error: enum 'AnchorType' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
   13 | enum class AnchorType : std::uint8_t { Static, Dynamic, All };
      |            ^

/Users/jviotti/Projects/core/src/core/jsonschema/include/sourcemeta/core/jsonschema_frame.h:109:14: error: enum 'Mode' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
  109 |   enum class Mode { Locations, References, Instances };
      |              ^

/Users/jviotti/Projects/core/src/core/jsonschema/include/sourcemeta/core/jsonschema_frame.h:144:14: error: enum 'LocationType' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
  144 |   enum class LocationType : std::uint8_t {
      |              ^

/Users/jviotti/Projects/core/src/core/jsonschema/include/sourcemeta/core/jsonschema_types.h:41:12: error: enum 'SchemaIdentificationStrategy' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
   41 | enum class SchemaIdentificationStrategy : std::uint8_t {
      |            ^

/Users/jviotti/Projects/core/src/core/jsonschema/include/sourcemeta/core/jsonschema_types.h:51:12: error: enum 'SchemaReferenceType' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
   51 | enum class SchemaReferenceType : std::uint8_t { Static, Dynamic };
      |            ^

/Users/jviotti/Projects/core/src/core/jsonschema/include/sourcemeta/core/jsonschema_types.h:62:12: error: enum 'SchemaKeywordType' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
   62 | enum class SchemaKeywordType : std::uint8_t {
      |            ^

/Users/jviotti/Projects/core/src/core/jsonschema/walker.cc:7:12: error: enum 'SchemaWalkerType_t' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
    7 | enum class SchemaWalkerType_t : std::uint8_t { Deep, Flat };
      |            ^

/Users/jviotti/Projects/core/src/core/regex/include/sourcemeta/core/regex.h:70:12: error: enum 'RegexIndex' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
   70 | enum class RegexIndex : std::uint8_t {
      |            ^

/Users/jviotti/Projects/core/src/extension/alterschema/include/sourcemeta/core/alterschema.h:25:12: error: enum 'AlterSchemaCategory' uses a larger base type ('int', size: 4 bytes) than necessary for its value set, consider using 'std::uint8_t' (1 byte) as the base type to reduce its size [performance-enum-size,-warnings-as-errors]
   25 | enum class AlterSchemaCategory {
      |            ^

```

--- a/cmake/common/targets/clang-tidy.config
+++ b/cmake/common/targets/clang-tidy.config
@@ -8,7 +8,7 @@ Checks: '-*,
           modernize-*,
           concurrency-*,
           cppcoreguidelines-*,
-          performance-*,
+          performance-*,-performance-enum-size,
           portability-*,
           objc-*,
           misc-*,-misc-no-recursion'

